src/main/python/rlbot/agents/hivemind/drone_agent.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+from rlbot.agents.base_independent_agent import BaseIndependentAgent
+from rlbot.botmanager.helper_process_request import HelperProcessRequest
+
+class DroneAgent(BaseIndependentAgent):
+    # Path to the hivemind helperprocess python file.
+    hive_path = None
+    # Bots with the same key will be part of the same hivemind.
+    hive_key = None
+    # Name of your hivemind that shows up in the console.
+    hive_name = None
+
+    def __init__(self, name, team, index):
+        super().__init__(name, team, index)
+        if self.hive_path is None:
+            raise NotImplementedError('You need to specify a path to the hivemind file.')
+        if self.hive_key is None:
+            raise NotImplementedError('You need to specify a key for your hivemind.')
+        if self.hive_name is None:
+            raise NotImplementedError('You need to specify a name for your hivemind.')
+    
+    def run_independently(self, terminate_request_event):
+        pass
+
+    def get_helper_process_request(self) -> HelperProcessRequest:
+        if os.path.isfile(self.hive_path):
+            # Appends team to key so that each team has its own hivemind.
+            key = f'{self.hive_key}{self.team}'
+            
+            # Creates request for helper process.
+            options = {
+                'name': self.hive_name
+            }
+
+            return HelperProcessRequest(self.hive_path, key, options=options)
+
+        raise FileNotFoundError(f'Could not find file: {self.hive_path}')
```

Comment:
May want to also append the hive_path onto the key. Without that, it's possible (but unlikely) that two different botmakers could use the same key, be on the same team, and accidentally get routed to the same process.

Also, I have a small preference to have the code be like 
```
if not os.path.isfile(self.hive_path):
    raise FileNotFoundError(f'Could not find file: {self.hive_path}')
```
So that there's less indentation and therefore less effort to keep track of what's going on when reading the code. You don't have to change it if this is your preferred style though.

Reply:
Done.